File: frontend/inventory-app/src/features/transactions/data/interfaces/Transaction.ts
export type TransactionType = 'Compra' | 'Venta';

export interface Transaction {
  id: number;
  date: string;
  type: TransactionType;
  productId: number;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  details: string;
}

export interface TransactionFormData {
  type: TransactionType;
  productId: number;
  quantity: number;
  unitPrice: number;
  details: string;
}

export interface TransactionFilters {
  startDate?: string;
  endDate?: string;
  type?: TransactionType;
  productId?: number;
} 

File: backend/TransactionService/Models/Transaction.cs
namespace TransactionService.Models
{
    public class Transaction
    {
        public int Id { get; set; }
        public DateTime Date { get; set; } = DateTime.UtcNow;
        public string Type { get; set; } = "";
        public int ProductId { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal TotalPrice => Quantity * UnitPrice;
        public string? Details { get; set; }
    }
}
